src/model/log_prob.jl

```diff
@@ -0,0 +1,443 @@
+"""
+function colors_to_fluxes(brightness::Float64, colors::Vector{Float64})
+    # build up log fluxes
+    ret    = Array(Float64, 5)
+    lnr    = brightness
+    ret[3] = lnr     # r flux
+    ret[4] = lnr + colors[3]        # ln(i/r) = c3 => lni = lnr - c3
+    ret[5] = ret[4] + colors[4]     # ln(z/i) = c4 => lnz = lni - c4
+    ret[2] = -colors[2] + lnr       # ln(r/g) = c2 => lng = c2 + lnr
+    ret[1] = -colors[1] + ret[2]    # ln(g/u) = c1 => lnu = c1 + lng
+    return exp(ret)
+end
+
+
+#######################################################################
+# galaxy shape transformation (unconstrained => constrained and back) #
+#######################################################################
+
+
+function constrain_gal_shape(unc_gal_shape::Vector{Float64})
+    gdev, gaxis, gangle, gscale = unc_gal_shape
+    constr_shape    = Array(Float64, 4)
+    constr_shape[1] = clamp(sigmoid(gdev), EPS, 1-EPS)
+    constr_shape[2] = clamp(sigmoid(gaxis), EPS, 1-EPS)
+    constr_shape[3] = gangle       # TODO put this between [0, 2pi]
+    constr_shape[4] = clamp(exp(gscale), EPS, Inf)
+    return constr_shape
+end
+
+
+function unconstrain_gal_shape(con_gal_shape::Vector{Float64})
+    gdev, gaxis, gangle, gscale = con_gal_shape
+    unc_shape    = Array(Float64, 4)
+    unc_shape[1] = logit(gdev)
+    unc_shape[2] = logit(gaxis)
+    unc_shape[3] = gangle
+    unc_shape[4] = log(gscale)
+    return unc_shape
+end
+
+
+######################################################
+# initialize from catalog and variational parameters #
+######################################################
+
+
+function init_star_state(entry::CatalogEntry)
+    brightness, colors = fluxes_to_colors(entry.star_fluxes)
+    param_vec = [brightness; colors; entry.pos]
+end
+
+
+function init_galaxy_state(entry::CatalogEntry)
+    brightness, colors = fluxes_to_colors(entry.gal_fluxes)
+    #gdev, gaxis, gangle, gscale = gal_shape
+    gal_shape = unconstrain_gal_shape([
+                    clamp(entry.gal_frac_dev, EPS, 1.-EPS),
+                    clamp(entry.gal_ab, EPS, 1.-EPS),
+                    entry.gal_angle,
+                    clamp(entry.gal_scale, EPS, Inf)
+                    ])
+    param_vec = [brightness; colors; entry.pos; gal_shape]
+    return param_vec
+end
+
+
+function elbo_args_vp_to_star_state(vp::Array{Float64,1})
+    fluxes      = variational_params_to_fluxes(vp)
+    lnr, colors = fluxes_to_colors(fluxes)
+    ra, dec     = vp[ids.u[1]], vp[ids.u[2]]
+    return [[lnr]; colors; [ra, dec]]
+end
+
+
+function elbo_args_vp_to_galaxy_state(vp::Array{Float64, 1})
+    shared_params = elbo_args_vp_to_star_state(vp)
+    gal_shape = [vp[ids.e_dev]  ,
+                 vp[ids.e_axis] ,
+                 vp[ids.e_angle],
+                 vp[ids.e_scale]]
+    return [shared_params; unconstrain_gal_shape(gal_shape)]
+end
+
+
+function variational_params_to_fluxes(i::Int, vs::Array{Array{Float64,1}})
```

Comment:
Can you initialize this from the `CatalogEntry` variables rather than the variational parameters? We really should move all this `vp`/`vs` stuff to within the `DeterministicVI` module.